lib/client/redirect.go

```diff
@@ -145,15 +145,15 @@ func (rd *Redirector) Start() error {
 	query.Set("secret_key", rd.key.String())
 	u.RawQuery = query.Encode()
 
-	out, err := rd.proxyClient.PostJSON(rd.context, rd.proxyClient.Endpoint("webapi", rd.Protocol, "login", "console"), SSOLoginConsoleReq{
+	out, err := rd.proxyClient.PostJSONWithFallback(rd.context, rd.proxyClient.Endpoint("webapi", rd.Protocol, "login", "console"), SSOLoginConsoleReq{
 		RedirectURL:       u.String(),
 		PublicKey:         rd.PubKey,
 		CertTTL:           rd.TTL,
 		ConnectorID:       rd.ConnectorID,
 		Compatibility:     rd.Compatibility,
 		RouteToCluster:    rd.RouteToCluster,
 		KubernetesCluster: rd.KubernetesCluster,
-	})
+	}, true)
```

Comment:
This should not always be `true`, it should come from the user setting the `--insecure` flag.

Reply:
That's probably correct. I vaguely remember not having access to the value of that flag at the time and intending to come back to this later.